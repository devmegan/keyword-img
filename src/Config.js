const HREFS = {
    app: 'https://devmegan.github.io/keyword-img/',
    repo: 'https://github.com/devmegan/keyword-img',
  }
  
const THEME = "garden";

const DESCRIPTION = {
    header: 'Enter a keyword...', 
    para: 'To receive an image associated to it'
}

export { HREFS, THEME, DESCRIPTION };
