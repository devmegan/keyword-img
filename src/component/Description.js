import { DESCRIPTION } from '../Config'

export default function Description() {
    return (
        <>
            <h1 className="text-2xl font-bold">{DESCRIPTION.header}</h1>
            <p>{DESCRIPTION.para}</p>
        </>
    )
}