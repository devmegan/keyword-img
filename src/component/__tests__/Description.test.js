import { render, screen } from '@testing-library/react';
import Description from '../Description';
import { DESCRIPTION } from '../../Config'

describe('description', () => {
    it('renders w/o crashing', () => {
        expect.hasAssertions();

        const view = render(<Description />);

        expect(view).toMatchSnapshot();
    });

    it('renders correct content from config', () => {
        expect.hasAssertions();

        render(<Description />)

        expect(screen.getByText(
            DESCRIPTION.para, DESCRIPTION.header
        )).toBeInTheDocument();
    });
});