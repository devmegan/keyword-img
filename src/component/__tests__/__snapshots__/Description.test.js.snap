// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`description renders w/o crashing 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h1
        class="text-2xl font-bold"
      >
        Enter a keyword...
      </h1>
      <p>
        To receive an image associated to it
      </p>
    </div>
  </body>,
  "container": <div>
    <h1
      class="text-2xl font-bold"
    >
      Enter a keyword...
    </h1>
    <p>
      To receive an image associated to it
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
